solr/core/src/java/org/apache/solr/handler/component/ShardRequestor.java

```diff
@@ -129,7 +129,12 @@ public ShardResponse call() throws Exception {
         // all of the servers for a shard are down.
         throw new SolrException(SolrException.ErrorCode.SERVICE_UNAVAILABLE, "no servers hosting shard: " + shard);
       }
-
+      if (sreq.requestStartTime > 0) {
```

Comment:
just a very minor nit - should we name this to `requestSubmitTime` instead? since within this `ShardRequestor` there's another local variable `startTime`, which indicates the start time of actual making such request. 

Or we can simply put a comment here mentioning that `sreq.requestStartTime` here refers to time the request is constructed and submitted to the ShardHandler? (ie submitted to the queue, the task not yet started) so the duration here measures the time delay in the queue/threadpool?

Reply:
Updated comment!